src/components/BookList.js

```diff
@@ -1,8 +1,16 @@
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+import { fetchBooks } from '../redux/books/booksSlice';
 import Book from './Book';
 import AddBook from './AddBook';
 
 const BooksList = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
   const books = useSelector((state) => state.books);
   return (
```

Comment:
- [ ] Great job on your progress in connecting your bookstore to an API in this milestone, enabling functionalities to get, delete, and post data. Your efforts are commendable. However, I'd like to highlight a few issues that could further improve your project.

   - One issue I encountered while reviewing your project is related to the use of `useSelector` to retrieve all books from your Redux store. During testing, I encountered an error indicating that `books.map` is not a function, which is caused by `books` being **undefined**.

    - To resolve this issue and ensure that you can access the `books` array from the Redux store, I recommend destructuring `books` directly from the Redux state instead of defining it as a separate variable. Instead of:

```javascript
const books = useSelector((state) => state.books);
```

You should use:

```javascript
const { books } = useSelector((state) => state.books);
```

  - This change ensures that you're correctly extracting the `books` array from your Redux state, which should resolve the error and improve the stability of your application.

Happy coding! 😊🚀🍀